setup.py

```diff
@@ -114,6 +114,8 @@
 
 POSTGRES_REQUIRED = [
     "psycopg2-binary>=2.8.3,<3",
+    "adbc-driver-manager==0.6.0",
```

Comment:
Not every one using Postgres as store needs this. I wonder if there is another way to handle the dependency

Reply:
It'll probably make sense to rewrite other methods to use adbc as well in the future, so i don't think it's worth it to split up extras now.

Reply:
do all the offline data sources support the adbc now?

Reply:
No, just postges and sqlite in the [docs](https://arrow.apache.org/adbc/current/python/index.html). I thought about this some more and using adbc in some methods of the same class and `psycopg2` in others is probably not the best solution anyway. While the feature would be useful, since the author is no longer active here, I think we should close this one. @jeremyary